server/src/main/java/org/apache/kafka/server/ClientMetricsManager.java

```diff
@@ -493,4 +520,124 @@ public void run() {
             }
         }
     }
+
+    // Visible for testing
+    final class ClientMetricsStats {
+
+        private static final String GROUP_NAME = "ClientMetrics";
+
+        // Visible for testing
+        static final String INSTANCE_COUNT = "ClientMetricsInstanceCount";
```

Comment:
So far, the naming convention for Kafka metrics is to use hyphen. So, ClientMetricsInstanceCount should be client-metrics-instance-count. Ditto below. Since the code hasn't been released yet. Could we make the change and update the KIP?

Reply:
@junrao Thanks for bringing this up. I did look into the naming for metrics and found it's kind of mixed in Kafka as of now. I have attached below 2 screenshots from what I see in JMX from `kafka.server` metrics itself. I myself was thinking that if there is some guidelines for the new metrics we follow.

<img width="425" alt="Screenshot 2024-02-22 at 10 41 08 PM" src="https://github.com/apache/kafka/assets/2861565/8435d2ff-a4cc-40e0-93c2-e06ee317fb4f">
<img width="341" alt="Screenshot 2024-02-22 at 10 41 16 PM" src="https://github.com/apache/kafka/assets/2861565/0363ff09-0dc1-48d6-80de-c112a4da3ba1">

Reply:
@apoorvmittal10 : On the server side, we have been using both Yammer metrics (org.apache.kafka.server.metrics.KafkaMetricsGroup) and Kafka metrics (org.apache.kafka.common.metrics.Metrics). The convention is that for the former, we use camel case. For the latter, we use hyphen.

Reply:
@junrao thanks for the details. I have opened the PR to address same: https://github.com/apache/kafka/pull/15680
cc: @AndrewJSchofield